pkg/cmd/server/kubernetes/master.go

```diff
@@ -69,11 +71,42 @@ const (
 // a single string value).
 func (c *MasterConfig) InstallAPI(container *restful.Container) ([]string, error) {
 	c.Master.RestfulContainer = container
-	_, err := master.New(c.Master)
+
+	// keep the old value
+	enableCoreControllers := c.Master.EnableCoreControllers
+
+	// explicitly set to false because master.New will start the bootstrap controller otherwise, and
+	// we want to start it ourselves so we can possibly override the EndpointReconciler
+	c.Master.EnableCoreControllers = false
+
+	m, err := master.New(c.Master)
 	if err != nil {
 		return nil, err
 	}
 
+	// restore the original value
+	c.Master.EnableCoreControllers = enableCoreControllers
+
+	if enableCoreControllers {
+		bootstrapController := m.NewBootstrapController()
```

Comment:
This isn't a maintainable factoring.  We'll be risking a slip every rebase for how they're handling the meaning of `enableCoreControllers`.  If we need a twiddle knob into how the `EndpointReconciler` is set, we need to refactor `master.go` upstream sufficiently to be confident that we don't go out of sync in a rebase.